tools/src/apply_acme_cert.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::Result;
+use clap::Parser;
+use warp::Filter;
+
+use crate::hyper_fetcher::HyperFetcher;
+use crate::linux_commands::{create_certificate_request_pem, create_private_key_pem};
+
+#[derive(Debug, Parser)]
+pub struct Opts {
+    #[clap(long)]
+    port: u16,
+    /// Directory URL of ACME server
+    #[clap(long)]
+    acme_server: String,
+    #[clap(long)]
+    email: String,
+    #[clap(long)]
+    domain: String,
+}
+
+fn start_warp_server(port: u16, answer: String) -> tokio::sync::oneshot::Sender<()> {
+    let (tx, rx) = tokio::sync::oneshot::channel();
+    let routes =
+        warp::path!(".well-known" / "acme-challenge" / String).map(move |_name| answer.to_string());
+    let (_addr, server) =
+        warp::serve(routes).bind_with_graceful_shutdown(([127, 0, 0, 1], port), async {
+            rx.await.ok();
+        });
+    tokio::spawn(server);
+    tx
+}
+
+pub async fn main(opts: Opts) -> Result<()> {
+    let acme_private_key = {
+        let private_key_file = "acme_account_private_key.pem";
```

Comment:
The names of these files should be in a config/TOML.  Ditto below.

Reply:
Added them as CLI args with default value.